### DISCUSSION TITLE
Geometry Cache option

### DISCUSSION THREAD

**Discussion by coolmuzzy** [masqu3rad3/tik_manager4#75] [Ideas]
Hi masqu3rad3, I hope you're doing well. I'm a Maya user and I've been using your product, Tik Manager, which is great. I have a suggestion for you.

Could you add a geometry cache option similar to Alembic export? Here's how it would work:

In Maya, when rigging, I create geometry meshes and organize them into selection sets like face, hands, and body. Then, I combine these meshes and export a geometry cache.

Simultaneously, I do the same thing in the shaded model, using the same selection sets, combine them, and import the geometry cache.

Having these kinds of options would greatly benefit us in production.

**Comment by masqu3rad3** [comment]
Hi coolmuzzy,

Although I am not a big fan of geometry caches, that sounds like an efficient workflow.

What you ask can be done easily. Moreover, that can be modified/tailored to make your workflow even smoother.
Can you provide me a little bit more info about this:

1. Silly question but my lack of expertise with other DCCs may cause miscommunication. We are talking about Maya right? :)
2. Do you have a script/snippet or another tool that you are using to geocache export? I cannot recall if there were any settings for geo caches but would like to know if there are any special checks needed for the workflow
3. Do you you export all caches in one go? Do you have any other selection sets in the scene that you do NOT want to export?

I try to avoid scene selection-based exports (where the user selects the nodes manually and exporter collects them) as much as possible since they are not programmatically repeatable and have a much larger human error margin. Therefore, to make the extract generic, I can make one exporting ALL meshes or exporting ALL selection sets.

Having that said, Extract, Validate and Ingest mechanism is something flexible and we can make a special extract plugin for your speciific workflow. For example, if you are following a certain naming convention for your sets, the publisher can collect those and export them only. Additionally, it is very easy to add a validator to check the existence of these sets before publishing to save some time and give proper guidance to artists in the team.

Let me know what you think.
Best

**Comment by coolmuzzy** [comment]
1) Yes, I am talking about Maya.
2) Yes, I do have a script. I will share a screen recording so that you can have an idea.
3) No, We have the selection checklist, and we just check the ones we want to export.

**Comment by coolmuzzy** [comment]
Hi,
 
Hope you are doing well!
Sorry for the delay I got busy due to some production overload.
 
I checked out your recent update. It is very well improved and much useful options also been added like usds etc. I really appreciate the efforts you are putting in this.
 
I recorded the quick demo of how the geometry cache workflow works in our pipeline. Whenever you have time please do check and let me know if any further info required. Geometry cache is actually a great workflow and lot of companies are actually using it professionally.

Here is the youtube link.
https://youtu.be/Kaf_smnclac

**Comment by masqu3rad3** [comment]
Thank you for the info and video. I watched it very quickly, I will watch it again and give it a deeper thought when I have more time.

My only initial concern here is combining...

With a simple geo cache set with only 2 elements, it is fine. But imagine that there is a character with full costume, body, nails, hat, shirt, buttons, shoes, shoelaces etc... 

1.  The combining will be a very time consuming process and it will probably crash the scene on larger poly counts.
2.  It will be getting harder and harder to make sure we are selecting with the right order. If selection order is wrong, it may create issues hard to debug.

So my counter offer is, why not a cache per object? Tik Manager can automate the pairing and assign it to the correct piece, so on user and it will still look like a single file but under the hood there will be as many caches as the count of the geos.
